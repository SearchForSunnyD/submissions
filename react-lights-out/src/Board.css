.Board {
  margin: 0 auto;
  border: black solid 2px;
  border-radius: 20px;
  padding: 30px;
  background-color: khaki;
}
