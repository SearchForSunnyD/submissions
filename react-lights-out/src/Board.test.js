import { fireEvent, render } from "@testing-library/react";
import React from "react";
import Board from "./Board";

it("renders without crashing", () => {
  render(<Board />);
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <Board nrows={1} ncols={1} chanceLightStartsOn={1} />,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays a win if won", () => {
  const { container } = render(
    <Board nrows={1} ncols={1} chanceLightStartsOn={0} />,
  );
  expect(container.querySelector("h1")).toBeInTheDocument();
});

it("updates cell", () => {
  const { container } = render(
    <Board nrows={3} ncols={3} chanceLightStartsOn={1} />,
  );

  const firstCell = container.querySelectorAll("td.Cell");

  expect(firstCell[0].className).toBe('Cell Cell-lit');

  fireEvent.click(firstCell[0]);

  expect(firstCell[0].className).toBe("Cell ");
});
