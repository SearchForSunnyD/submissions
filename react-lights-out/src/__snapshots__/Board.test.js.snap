// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <table
    class="Board"
  >
    <tbody>
      <tr>
        <td
          class="Cell Cell-lit"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
